code/network/multiutil.cpp

```diff
@@ -1861,19 +1860,27 @@ int multi_can_message(net_player *p)
 
 	// only wing/team leaders can message
 	case MSO_SQUAD_LEADER:
+	{
 		// if the player has an invalid object #
 		if(p->m_player->objnum < 0){
 			return 0;
 		}
 
 		// check to see if he's a wingleader
-		sp = &Ships[Objects[p->m_player->objnum].instance];
-		if (sp->ship_name[strlen(sp->ship_name)-1] == '1') 
+		const ship_registry_entry* ship_regp = ship_registry_get(Ships[p->m_player->objnum].ship_name);
```

Comment:
One thing I missed on my first go-round... `objnum` is an index into `Objects`, not into `Ships`.  You will need to get the object's `instance` field, as in the deleted code: `Ships[Objects[p->m_player->objnum].instance]`

Reply:
Yeah, I'm not sure how I missed that, either.  Anyway, that should be fixed now.  Let me know if it needs more changes.